lib/active_merchant/billing/gateways/authorize_net_cim.rb

```diff
@@ -216,9 +216,13 @@ def delete_customer_shipping_address(options)
       #
       # ==== Options
       #
-      # * <tt>:customer_profile_id</tt> -- The Customer Profile ID of the customer to retrieve. (REQUIRED)
+      # * <tt>:customer_profile_id</tt> -- The Customer Profile ID of the customer to retrieve. (CONDITIONAL)
+      # * <tt>:email</tt> -- The Email address associated with the customer to retrieve. (CONDITIONAL)
+      # * <tt>:merchant_customer_id</tt> -- The Merchant assigned ID of the customer to retrieve. (CONDITIONAL)
       def get_customer_profile(options)
-        requires!(options, :customer_profile_id)
+        requires!(options, :customer_profile_id) unless options[:email] || options[:merchant_customer_id]
```

Comment:
I think this would be better as a straight conditional, rather than three guarded `requires!` calls. Best as I can figure, what you're trying to do is say you want one of `customer_profile_id`, `email`, or `merchant_customer_id`. In that case, I'd skip `requires!` entirely (which is just a function that throws `ArgumentError` in certain conditions) and throw the exception yourself, or do simpler logic. As an example, something like (didn't think through this, not tested, may be wrong, but points in the direction I'm thinking)

`raise ArgumentError("Requires one of customer_profile_id, email, or merchant_customer_id") unless options[:customer_profile_id] || options[:email] || options[:merchant_customer_id]`

might be a better fit.

Reply:
Sounds good, I was just following a pattern I found in `create_customer_profile`. But that would be a lot cleaner.